### PR TITLE
fix: Warn users when GitHub Actions is making a merge commit

### PR DESCRIPTION
**Pull request by srijan-deepsource**
changes:
  - Change signature of `gitGetHead` to return a warning for GitHub Actions merge commits
  - Propagate the warning to `report` command response

### CODE REVIEW

File: command/report/git.go [hunk 1]
```diff
@@ -80,3 +75,59 @@ func fetchHeadManually(directoryPath string) (string, error) {
...
+	return topCommit, warning, err
+}
+
+// Return PR's HEAD ref set as env variable manually by DeepSource's Test coverage action.
+func getTestCoverageActionCommit() (headOID string, err error) {
+	// This is kept separate from `getGitHubActionsCommit` because we don't want to run any git command manually
+	// before this is checked. Since this is guaranteed to be set if artifact is sent using our GitHub action,
+	// we can reliably send the commit SHA, and no git commands are executed, making the actions work all the time. \o/
+
+	// We are setting PR's head commit as default using github context as env variable: "GHA_HEAD_COMMIT_SHA"
```
Comment (srijan-deepsource) [inline command/report/git.go:116]: https://github.com/DeepSourceCorp/test-coverage-action/pull/29/files#diff-b10564ab7d2c520cdd0243874879fb0a782862c3c902ab535faabe57d5a505e1R52

File: command/report/git.go [hunk 2]
```diff
@@ -11,29 +11,31 @@ import (
 )
 
 // gitGetHead accepts a git directory and returns head commit OID / error
-func gitGetHead(workspaceDir string) (string, error) {
+func gitGetHead(workspaceDir string) (headOID string, warning string, err error) {
+	// Check if DeepSource's Test coverage action triggered this first before executing any git commands.
+	headOID, err = getTestCoverageActionCommit()
+	if headOID != "" {
+		return headOID, warning, err
```
Comment (sourya-deepsource) [inline command/report/git.go:18]:
```suggestion
		return
```

File: command/report/git.go [hunk 3]
```diff
@@ -11,29 +11,31 @@ import (
...
+	}
+
+	// get the top commit manually, using git command
+	headOID, err = fetchHeadManually(workspaceDir)
+	if err != nil {
+		fmt.Println(err)
+		sentry.CaptureException(err)
+		return "", "", err
```
Comment (sourya-deepsource) [inline command/report/git.go:26]:
```suggestion
		return
```

File: command/report/git.go [hunk 4]
```diff
@@ -47,18 +49,11 @@ func gitGetHead(workspaceDir string) (string, error) {
...
-	// nor a travis env with PR. Continue to fetch the headOID via the git command.
-	headOID, err := fetchHeadManually(workspaceDir)
-	if err != nil {
-		fmt.Println(err)
-		sentry.CaptureException(err)
-		return "", err
-	}
-	return headOID, nil
+	// If we are here, it means there weren't any special cases. Return the manually found headOID.
+	return headOID, warning, nil
```
Comment (sourya-deepsource) [inline command/report/git.go:56]:
```suggestion
	return
```

**Review by sourya-deepsource — CHANGES_REQUESTED** [review]

**Review by sourya-deepsource — APPROVED** [review]

**Pull request merged**